File: app/src/main/java/org/ligi/ipfsdroid/AppModule.java
package org.ligi.ipfsdroid;

import dagger.Module;
import dagger.Provides;
import io.ipfs.kotlin.IPFS;
import javax.inject.Singleton;
import okhttp3.OkHttpClient;

@Module
public class AppModule {

    private final App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Singleton
    @Provides
    OkHttpClient proviceOkhttp() {
        return new OkHttpClient.Builder().build();
    }

    @Singleton
    @Provides
    IPFS provideIPFS() {
        return new IPFS();
    }

    @Singleton
    @Provides
    State provideState() {
        return new State();
    }
}

File: app/src/main/java/org/ligi/ipfsdroid/di/AppModule.kt
package org.ligi.ipfsdroid.di

import android.content.Context
import android.support.annotation.NonNull
import dagger.Module
import dagger.Provides
import io.ipfs.kotlin.IPFS
import okhttp3.OkHttpClient
import org.ligi.ipfsdroid.repository.Repository
import java.util.concurrent.TimeUnit
import javax.inject.Singleton

@Module
class AppModule(val context: Context) {

    @Singleton
    @Provides
    @NonNull
    fun provideContext() : Context {
        return context
    }

    @Singleton
    @Provides
    internal fun provideOkhttp() = OkHttpClient.Builder().let { builder ->
        builder.connectTimeout(1000, TimeUnit.SECONDS)
        builder.readTimeout(1000, TimeUnit.SECONDS)
        builder.build()
    }

    @Singleton
    @Provides
    internal fun provideIPFS(providedOkHttp: OkHttpClient)
            = IPFS(okHttpClient = providedOkHttp)

    @Singleton
    @Provides
    internal fun provideRepository(ipfs: IPFS) = Repository(ipfs = ipfs)

}
